Adding support for the extra parameters that can be used to authenticate via devise

brightbits opened the pull request:
Closes #19

baschtl commented on lib/devise/token_authenticatable/model.rb:
```diff
@@ -32,7 +32,12 @@ module TokenAuthenticatable
 
         def find_for_token_authentication(conditions)
-          find_for_authentication(authentication_token: conditions[Devise::TokenAuthenticatable.token_authentication_key])
+          conditions = conditions.dup
```
Please, don't overwrite a method argument. Just name the variable differently, e.g., `auth_conditions`.

brightbits replied:
Changing

baschtl commented on spec/models/devise/token_authenticatable/model_spec.rb:
```diff
@@ -51,6 +51,16 @@
+        expect(described_class).to receive(:find_for_authentication).with(expected_conditions).and_call_original
+
+        authenticated_entity = described_class.find_for_token_authentication(conditions)
+        expect(entity.authentication_token).to eq(authenticated_entity.authentication_token)
```
Please, only use one expectation per example as this makes debugging easier. Write another example for the second expectation.

baschtl commented:
Beyond the mentioned things this is good to merge. Thanks a lot!

brightbits commented:
Renamed the dupped parameter, removed the second expectation, it wasn't actually testing anything beyond that code outside the gems control did what I wanted it to, all the test cares about is that the extra parameters make it to the `find_for_authentication` method.

baschtl commented:
@brightbits Thanks again. I will release a new version as soon as possible.

brightbits commented:
Cheers :+1:
